src/lib/entropy/darwin_secrandom/darwin_secrandom.cpp

```diff
@@ -0,0 +1,28 @@
+/*
+* Darwin SecRandomCopyBytes EntropySource
+* (C) 2015 Daniel Seither (Kullo GmbH)
+*
+* Botan is released under the Simplified BSD License (see license.txt)
+*/
+
+#include <botan/internal/darwin_secrandom.h>
+#include <Security/Security.h>
+
+namespace Botan {
+
+/**
+* Gather entropy from SecRandomCopyBytes
+*/
+void Darwin_SecRandom::poll(Entropy_Accumulator& accum)
+   {
+   const size_t ENTROPY_BITS_PER_BYTE = 8;
+   const size_t BUF_SIZE = 256;
+
+   m_buf.resize(BUF_SIZE);
+   if (0 == SecRandomCopyBytes(kSecRandomDefault, m_buf.size(), m_buf.data()))
```

Comment:
Sounds right to me. Maybe a call failed now for some transient reason, but would work if we called it again a few msecs from now. Considering a failure here as simply 'this poll attempt on this source produced no usable entropy' is likely the easiest way to reason about it, especially as the failure modes here are unclear, and whatever they may be there is not much we can do about them from userspace.